src/trio/_tests/test_socket.py

```diff
@@ -512,12 +512,12 @@ def gai_without_v4mapped_is_buggy() -> bool:  # pragma: no cover
         return True
 
 
-@attr.s
+@attrs.define(slots=False)
 class Addresses:
-    bind_all: str = attr.ib()
-    localhost: str = attr.ib()
-    arbitrary: str = attr.ib()
-    broadcast: str = attr.ib()
+    bind_all: str = attrs.field()
+    localhost: str = attrs.field()
+    arbitrary: str = attrs.field()
+    broadcast: str = attrs.field()
```

Comment:
With the new attrs version, we don't need the `field()` calls if they don't have any options set. For tests, might as well also remove `slots=False`, since it can't cause backwards compatibility problems.

Reply:
Re `field`: orly? I did that transform for `attrs.define` and `attrs.frozen` decorators, but didn't realize it applied to non-decorator `attrs.field`, as well, but confirmed. :)

Re: `slots=False`, I'm not sure what you mean. `slots=False` was the default for `attr.s`, so all that didn't specify had it `False`. But with the new ones, `slots=True` is now the default, so I think all of those that didn't specify before now need `slots=False` to guarantee the same behavior. Can confirm that removing that _does_ immediately break a bunch of tests.

Reply:
Correction: It doesn't break the tests - my mistake. Can you enlighten me as to why this is actually safe to do? It seems to be that if they were slotted before, we want them to still not be slotted now.

Reply:
I deleted a comment here because I thought I'd made a mistake about that breaking the tests, but no, confirmed it does break them.

Reply:
It also turns out that while the bare `x = attrs.field` construct works at runtime, it makes both `pyright` and `mypy` very unhappy, and it fails CI. Apparently it's not supported by type-checkers yet. I've removed a3550387fc902f2cae61cb231e17a57ee07ac3d8 from the branch.

Reply:
For `field`, I was meaning using an annotation, no assignment at all. I was thinking that making them slotted would likely work in the tests, but I guess not.

Reply:
I would be in favor of enabling slots everywhere where it makes sense to do so

Reply:
Sure, but that seems like a different issue.

Reply:
@TeamSpen210 Ohh, of course! Now that you point it out, I don't know how I missed it. Type checkers have no issue with that, added 88473eaafac2350f65e1fd51824e0218183ad17e.